styx-scheduler-service/src/main/java/com/spotify/styx/TriggerManager.java

```diff
@@ -97,10 +99,11 @@ void tick() {
 
       if (enabled.contains(workflow.id())) {
         try {
-          triggerListener.event(
+          CompletionStage<Void> processed = triggerListener.event(
               workflow,
               Trigger.natural(),
               decrementInstant(next, partitioning));
+          processed.toCompletableFuture().get(1, TimeUnit.MINUTES);
```

Comment:
Waiting forever is not necessarily bad. If it takes really long time to complete, we probably have a super long queue, and firing new events could make it worse. However this does change current behaviour. I would like to get some words from @rouzwawi